test/WooVaultManager.test.ts

```diff
@@ -325,3 +330,129 @@ describe('WooVaultManager', () => {
+  let wooPP: Contract
+  let newWooPP: Contract
+
+  let mintUSDT = BigNumber.from(30000)
+
+  let onlyOwnerRevertedMessage: string
+  let onlyAdminRevertedMessage: string
+
+  before(async () => {
+    ;[owner, admin, user, vault] = await ethers.getSigners()
+    usdtToken = await deployContract(owner, TestToken, [])
+    wooToken = await deployContract(owner, TestToken, [])
+
+    wooAccessManager = (await deployContract(owner, WooAccessManagerArtifact, [])) as WooAccessManager
+    await wooAccessManager.setVaultAdmin(admin.address, true)
+    newWooAccessManager = (await deployContract(owner, WooAccessManagerArtifact, [])) as WooAccessManager
+
+    wooPP = await deployMockContract(owner, IWooPP.abi)
+    await wooPP.mock.quoteToken.returns(usdtToken.address)
+    await wooPP.mock.querySellQuote.returns(MOCK_REWARD_AMOUNT)
+    await wooPP.mock.sellQuote.returns(MOCK_REWARD_AMOUNT)
+
+    newWooPP = await deployMockContract(owner, IWooPP.abi)
+    await newWooPP.mock.quoteToken.returns(usdtToken.address)
+
+    wooVaultManager = (await deployContract(owner, WooVaultManagerArtifact, [
+        usdtToken.address,
+        wooToken.address,
+        wooAccessManager.address
+      ])) as WooVaultManager
+
+    await usdtToken.mint(owner.address, mintUSDT)
+
+    onlyOwnerRevertedMessage = 'InitializableOwnable: NOT_OWNER'
+    onlyAdminRevertedMessage = 'WooVaultManager: NOT_ADMIN'
+  })
+
+  it('Only admin able to setVaultWeight', async () => {
+    let weight = BigNumber.from(100);
+    expect(await wooAccessManager.isVaultAdmin(user.address)).to.eq(false)
+    await expect(wooVaultManager.connect(user).setVaultWeight(vault.address, weight)).to.be.revertedWith(
+      onlyAdminRevertedMessage
+    )
+
+    expect(await wooAccessManager.isVaultAdmin(admin.address)).to.eq(true)
+    await wooVaultManager.connect(admin).setVaultWeight(vault.address, weight)
+    expect(await wooVaultManager.vaultWeight(vault.address)).to.eq(weight)
+
+    weight = BigNumber.from(200)
+    await wooVaultManager.connect(owner).setVaultWeight(vault.address, weight)
+    expect(await wooVaultManager.vaultWeight(vault.address)).to.eq(weight)
+  })
+
+  it('Only admin able to distributeAllReward', async () => {
+    // @qc TODO Wrong code that need to be fixed
+    expect(await usdtToken.balanceOf(owner.address)).to.eq(mintUSDT)
+    let rewardAmount = BigNumber.from(100)
+    await usdtToken.connect(owner).approve(wooVaultManager.address, rewardAmount)
+    await wooVaultManager.connect(owner).addReward(rewardAmount)
+    expect(await usdtToken.balanceOf(wooVaultManager.address)).to.eq(rewardAmount)
+
+    expect(await wooAccessManager.isVaultAdmin(user.address)).to.eq(false)
+    await expect(wooVaultManager.connect(user).distributeAllReward()).to.be.revertedWith(
+      onlyAdminRevertedMessage
+    )
+
+    expect(await wooAccessManager.isVaultAdmin(admin.address)).to.eq(true)
+    await wooVaultManager.connect(admin).distributeAllReward()
```

Comment:
1) WooVaultManager Access Control
       Only admin able to distributeAllReward:
     Error: VM Exception while processing transaction: reverted with reason string 'TransferHelper::safeApprove: approve failed'
      at WooVaultManager.onlyAdmin (contracts/WooVaultManager.sol:85)
      at WooVaultManager.safeApprove (@uniswap/lib/contracts/libraries/TransferHelper.sol:7)
      at WooVaultManager.distributeAllReward (contracts/WooVaultManager.sol:154)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at runNextTicks (node:internal/process/task_queues:65:3)
      at listOnTimeout (node:internal/timers:526:9)
      at processTimers (node:internal/timers:500:7)
      at EthModule._estimateGasAction (node_modules/hardhat/src/internal/hardhat-network/provider/modules/eth.ts:425:7)
      at HardhatNetworkProvider.request (node_modules/hardhat/src/internal/hardhat-network/provider/provider.ts:108:18)
      at EthersProviderWrapper.send (node_modules/@nomiclabs/hardhat-ethers/src/internal/ethers-provider-wrapper.ts:13:20)

Reply:
fixed